guides/document_lifecycle.md

```diff
@@ -0,0 +1,269 @@
+# Document Lifecycle
+
+This guide covers OpenSearch Ruby Client API actions for Document Lifecycle. You'll learn how to create, read, update, and delete documents in your OpenSearch cluster. Whether you're new to OpenSearch or an experienced user, this guide provides the information you need to manage your document lifecycle effectively.
+
+## Setup
+
+Assuming you have OpenSearch running locally on port 9200, you can create a client instance
+with the following code:
+
+```rust
+let url = Url::parse("http://localhost:9200")?;
+let conn_pool = SingleNodeConnectionPool::new(url);
+let transport = TransportBuilder::new(conn_pool).disable_proxy().build()?;
+let client = OpenSearch::new(transport);
+```
+
+Next, create an index named `movies` with the default settings:
+
+```rust
+index = 'movies'
+client.indices().create(IndicesCreateParts::Index(index)).send().await?;
+```
+
+## Document API Actions
+
+### Create a new document with specified ID
+
+To create a new document, use the `create` or `index` API action. The following code creates two new documents with IDs of `1` and `2`:
+
+```rust
+client
+  .indices()
+  .create(IndexParts::IndexId(index, 1))
+  .body(json!({ "title": "Beauty and the Beast", "year": 1991 }))
+  .send()
+  .await?;
+
+client
+  .indices()
+  .create(IndexParts::IndexId(index, 2))
+  .body(json!(
+    { "title": "Beauty and the Beast - Live Action", "year":  2017 }))
+  .send()
+  .await?;
+```
+
+Note that the `create` action is NOT idempotent. If you try to create a document with an ID that already exists, the request will fail:
+
+```rust
+client
+  .indices()
+  .create(IndexParts::IndexId(index, 1))
+  .body(json!({"title": "Just Another Movie" }))
+  .send()
+  .await? {
+    Ok(_response) => {};
+    Err(e) => println!("{}", e);
+  };
```

Comment:
This is not valid Rust, the actual error from OpenSearch can be retrieved like so:
```suggestion
let err = client
  .indices()
  .create(IndexParts::IndexId(index, 1))
  .body(json!({"title": "Just Another Movie" }))
  .send()
  .await?
  .exception()
  .await?
  .unwrap();
println!("{:#?}", err);
```